symengine/tests/test_sympify.py

```diff
@@ -14,6 +15,27 @@ def test_sympify1():
     assert false == sympify(False)
 
 
+def test_S():
+    assert S(0) == Integer(0)
+    assert S(1) == Integer(1)
+    assert S(-1) == Integer(-1)
+    assert S(1)/2 == Rational(1, 2)
+    assert S.One is S(1)
+    assert S.Zero is S(0)
+    assert S.NegativeOne is S(-1)
+    assert S.Half is S(1)/2
+    assert S.Pi is pi
+    assert S.NaN is nan
+    assert S.Infinity is oo
+    assert S.NegativeInfinity is -oo
+    assert S.ComplexInfinity is zoo
+    assert S.Exp1 is E
+    assert S.ImaginaryUnit is I
+    assert S.GoldenRatio is GoldenRatio
+    assert S.true is true
+    assert S.false is false
```

Comment:
Because we're returning the `Integer` object created directly?
```
class One(Integer):
    def __new__(cls):
        cdef Basic r = Number.__new__(Integer)
        r.thisptr = <RCP[const symengine.Basic]>symengine.integer(1)
        return r
```
I tried changing this to:
```
class One(Integer):
    def __init__(Basic self):
        self.thisptr = <RCP[const symengine.Basic]>symengine.integer(1)
```
but that doesn't work either.